core/src/main/java/org/jruby/ir/runtime/IRRuntimeHelpers.java

```diff
@@ -624,12 +624,9 @@ public static void checkArity(ThreadContext context, StaticScope scope, Object[]
      * @param runtime the current runtime
      * @return whether to print IR
      */
-    public static boolean shouldPrintIR(Ruby runtime) {
-        boolean booting = runtime.isBooting();
-        boolean print = Options.IR_PRINT.load();
-        boolean printAll = Options.IR_PRINT_ALL.load();
-
-        return (print && !booting) || (booting && printAll);
+    public static boolean shouldPrintIR(final Ruby runtime) {
+        final boolean booting = runtime.isBooting();
+        return (Options.IR_PRINT.load() && !booting) || (booting && Options.IR_PRINT_ALL.load());
```

Comment:
I kinda like the original better since the condition reads nice.. "if print and not booting, or booting and print all..."